### PR TITLE
[DOCS] Navigation & Navigation2D

### PR DESCRIPTION
**Pull request by denisw**
I have noticed that the API Reference had documentation for `Navigation`, but not `Navigation2D`. So I tried to use the former to add a class description and some method documentation to the latter. I hope everything is factually correct, especially the description of what the `optimize` parameter of `get_simple_path()` does.

### CODE REVIEW

File: doc/classes/Navigation2D.xml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="Navigation2D" inherits="Node2D" category="Core" version="3.1">
 	<brief_description>
-		Class to assist with character navigation and pathfinding.
+		Node class that assists with 2D navigation and pathfinding.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: Saying "Node class" is redundant, plus the most important information belongs at the front.  The brief description appears next to the name in context so you just need a few words to indicate the purpose. "2D navigation and pathfinding node."

File: doc/classes/Navigation2D.xml [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 	</brief_description>
 	<description>
+		The [code]Navigation2D[/code] class is used for navigation and pathfinding in the 2D space. By default it will automatically collect all child [code]NavigationPolygon[/code] resources, but they can also be added on the fly with :ref:`navpoly_add()<class_Navigation2D_navpoly_add>`.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: Don't put code tags around class names.

File: doc/classes/Navigation2D.xml [hunk 3]
```diff
@@ -24,6 +26,7 @@
 			<argument index="0" name="to_point" type="Vector2">
 			</argument>
 			<description>
+				Determines the closest navigation point to the point passed and returns the owning [code]Object[/code] of the point's navigation polygon, usually a [code]NavigationPolygonInstance[/code]. If the polygon was added dynamically with [code]navpoly_add()[/code], the owner that was passed in that call is returned (or [code]null[/code] if no owner was passed).
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: Other classes should be ref links: [Object] and [NavigationPolygonInstance]
Methods should be linked too: [method navpoly_add]

File: doc/classes/Navigation2D.xml [hunk 4]
```diff
@@ -36,6 +39,7 @@
 			<argument index="2" name="optimize" type="bool" default="true">
 			</argument>
 			<description>
+				Calculates a path between two local-coordinate points and returns the result as a [code]PoolVector2Array[/code]. If the [code]optimize[/code] is [code]true[/code] (the default), string pulling will be applied to smoothen the calculated path.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: Remove "-" between local and coordinate.
If you mention [PoolVector2Array] it should be a link, although it's already listed above as the return value. 
Remove "the" before optimize
"smoothen" isnt' a word, change to "smooth"

Suggestion for first sentence:
Returns the path between two given points. Points are in local coordinate space.

File: doc/classes/Navigation2D.xml [hunk 5]
```diff
@@ -48,6 +52,7 @@
 			<argument index="2" name="owner" type="Object" default="null">
 			</argument>
 			<description>
+				Adds a [code]NavigationPolygon[/code] to the navigation node. Returns an ID that can be used to refer to the polygon when calling [code]navpoly_remove()[/code] or [code]navpoly_set_transform()[/code]. A [code]Transform2D[/code] that can be applied optionally applied to the polygon. Also, the polygon may be associated with an owning [code]Node[/code] (or [code]Object[/code]) for [code]get_closest_point_owner()[/code].
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: " A [code]Transform2D[/code] that can be applied optionally applied to the polygon." - You used "applied" twice - remove the first one. Remove "that". 

Link objects and methods.

File: doc/classes/Navigation2D.xml [hunk 6]
```diff
@@ -56,6 +61,7 @@
 			<argument index="0" name="id" type="int">
 			</argument>
 			<description>
+				Removes the [code]NavigationPolygon[/code] with the passed ID from the navigation node.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: "given" sounds better than "passed".

File: doc/classes/Navigation2D.xml [hunk 7]
```diff
@@ -16,6 +17,7 @@
 			<argument index="0" name="to_point" type="Vector2">
 			</argument>
 			<description>
+				Returns the closest navigation point to the point passed. Both points are interpreted as relative to the navigation node's local coordinate system.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: "to the point passed" -> "to the given point"

File: doc/classes/Navigation2D.xml [hunk 8]
```diff
@@ -66,6 +72,7 @@
 			<argument index="1" name="xform" type="Transform2D">
 			</argument>
 			<description>
+				Sets the [code]Transform2D[/code] applied to the [code]NavigationPolygon[/code] with the passed ID.
```
Comment (cbscribe) [inline doc/classes/Navigation2D.xml]: Applies a transform to the [NavigationPolygon] with the given ID.

**Review by cbscribe — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I made a few changes and suggestions. Overall recommendation: avoid extra unnecessary addition of words (see what I did there?). It's a temptation because it makes the text feel more "technical" but this is a reference. Descriptions should be brief and to the point.

**Comment by denisw** [comment]
@cbscribe Thank you for the thorough review! I agree with pretty much everything you said. I based my docs on the ones of `Navigation` which seems to validate several of the guidelines you gave here; I guess that needs some updating as well. 🙂 

I'll tweak the PR anddo a new push.

**Comment by cbscribe** [comment]
Yeah, you have to be careful about using existing docs if they haven't been updated yet either.

If you're up for it, you can move your new descriptions over to `Navigation` as well. :)

**Comment by denisw** [comment]
@cbscribe Done. Looking forward to your feedback. :)

**Review by cbscribe — APPROVED** [review]

**Comment by cbscribe** [comment]
Looks great! Are you going to update navigation too? If so, you can add it to this PR. Otherwise this is ready to merge.

**Comment by denisw** [comment]
Sure, I can do that!

**Comment by hpvb** [comment]
Cherry picked into 3.0.3

**Pull request merged**